src/CaptureClient/CaptureEventProcessor.cpp

```diff
@@ -550,12 +552,33 @@ void CaptureEventProcessorForListener::ProcessThreadNamesSnapshot(
   }
 }
 
+[[nodiscard]] static ThreadStateSliceInfo::WakeupReason toWakeupReason(
```

Comment:
The name is a bit ambiguous because it's translating from a `WakeupReason` to another. What about something like "FromGrpcWakeupReasonToInfoWakeupReason"?